### PR TITLE
feat(trim file system) Support Remove Snapshots During Filesystem Trim

### PR DESCRIPTION
**Pull request by smallteeths**
wangsiye/longhorn-ui:20599d3

https://github.com/longhorn/longhorn/issues/836

Signed-off-by: Siye Wang <siye@rancher.com>

### CODE REVIEW

File: src/routes/volume/detail/index.js [hunk 1]
```diff
@@ -324,6 +325,25 @@ function VolumeDetail({ snapshotModal, dispatch, backup, engineimage, eventlog,
...
+        },
+      })
+    },
+    trimFilesystem(record) {
+      if (record?.actions?.trimFilesystem) {
+        dispatch({
+          type: 'volume/trimFilesystem',
+          payload: {
+            params: record,
+            url: record?.actions?.trimFilesystem,
```
Comment (innobead) [inline src/routes/volume/detail/index.js:342]: nit: ` record.actions.trimFilesystem`

File: src/routes/volume/detail/VolumeInfo.js [hunk 2]
```diff
@@ -294,6 +294,11 @@ function VolumeInfo({ selectedVolume, snapshotModalState, engineImages, hosts, c
         </Tooltip>
         {selectedVolume.controllers ? selectedVolume.controllers.filter(item => item.instanceManagerName !== '').map(item => <div key={item.hostId} style={{ fontFamily: 'monospace', margin: '2px 0px' }}> <span style={{ backgroundColor: '#f2f4f5' }}> {item.instanceManagerName} </span></div>) : ''}
       </div>
+      <div className={styles.row}>
+        <span className={styles.label}> Allow snapshots removal during trim:</span>
+        <span style={{ marginRight: 5 }}>{selectedVolume.unmapMarkSnapChainRemoved ? selectedVolume.unmapMarkSnapChainRemoved : ''}</span>
+        {selectedVolume?.controllers && selectedVolume?.controllers[0] && `(${selectedVolume?.controllers[0]?.unmapMarkSnapChainRemovedEnabled})`}
+      </div>
```
Comment (innobead) [inline src/routes/volume/detail/VolumeInfo.js:301]: Why we show two values here (from volume and it's engine)? How will it look like?

**Review by innobead — APPROVED** [review]
In general, LGTM.

File: src/routes/volume/detail/VolumeInfo.js (hunk 2 above)
Comment (innobead) [reply src/routes/volume/detail/VolumeInfo.js]: Thanks for the explanation :+1:

File: src/routes/volume/detail/index.js (hunk 1 above)
Comment (smallteeths) [reply src/routes/volume/detail/index.js]: done

**Pull request merged**